web_src/js/features/comp/EditorMarkdown.js

```diff
@@ -0,0 +1,100 @@
+import {triggerEditorContentChanged} from './Paste.js';
+
+function handleIndentSelection(textarea, e) {
+  const selStart = textarea.selectionStart;
+  const selEnd = textarea.selectionEnd;
+  if (selEnd === selStart) return; // do not process when no selection
+
+  e.preventDefault();
+  const lines = textarea.value.split('\n');
+  const selectedLines = [];
+
+  let pos = 0;
+  for (let i = 0; i < lines.length; i++) {
+    if (pos > selEnd) break;
+    if (pos >= selStart) selectedLines.push(i);
+    pos += lines[i].length + 1;
+  }
+
+  for (const i of selectedLines) {
+    if (e.shiftKey) {
+      lines[i] = lines[i].replace(/^(\t| {1,2})/, '');
+    } else {
+      lines[i] = `  ${lines[i]}`;
+    }
+  }
+
+  // re-calculating the selection range
+  let newSelStart, newSelEnd;
+  pos = 0;
+  for (let i = 0; i < lines.length; i++) {
+    if (i === selectedLines[0]) {
+      newSelStart = pos;
+    }
+    if (i === selectedLines[selectedLines.length - 1]) {
+      newSelEnd = pos + lines[i].length;
+      break;
+    }
+    pos += lines[i].length + 1;
+  }
+  textarea.value = lines.join('\n');
+  textarea.setSelectionRange(newSelStart, newSelEnd);
+  triggerEditorContentChanged(textarea);
+}
+
+function handleNewline(textarea, e) {
+  const selStart = textarea.selectionStart;
+  const selEnd = textarea.selectionEnd;
+  if (selEnd !== selStart) return; // do not process when there is a selection
+
+  const value = textarea.value;
+
+  // find the current line
+  const lineStart = value.lastIndexOf('\n', selStart - 1) + 1;
+  let lineEnd = value.indexOf('\n', selStart);
+  lineEnd = lineEnd < 0 ? value.length : lineEnd;
+  let line = value.slice(lineStart, lineEnd);
+  if (!line) return; // if the line is empty, do nothing, let the browser handle it
+
+  // parse the indention
+  const indention = /^\s*/.exec(line)[0];
+  line = line.slice(indention.length);
+
+  // parse the prefixes: "1. ", "- ", "* ", "[ ] ", "[x] "
+  const prefixMatch = /^([0-9]+\.|[-*]|\[ \]|\[x\])\s/.exec(line);
```

Comment:
```suggestion
  const prefixMatch = /^([0-9]+\.|[-*]|\[ \]|\[x\])\s*/.exec(line);
```
should probably yield less false negatives

Reply:
I do not think so, please show a real case.

Reply:
It conforms with GitHubs' behavior.
GitHub behaves exactly like that.

Reply:
Could you elaborate with an example about what behavior? Do you mean something need to change, or current implementation is fine?

Reply:
?
I am confused.
What I mean is that we accept any amount of whitespace after the prefix.
That is also how GitHub handles it:
When you enter either `1.   ` or `1.` (no space), both times your list will still be auto-completed when pressing <kbd>Enter</kbd>.

Reply:
Okay, that is weird:
Did GitHub change their behavior since I last tried it yesterday, or did I hallucinate it working?
Right now, `1.a` no longer works for autocompleting a list.

Reply:
Actually, `1.a` never works for autocompleting a list ....... 

For example: `1.234` is a number, it is not for a list.